backend/api/src/api/v1/gql/chat.rs

```diff
@@ -0,0 +1,143 @@
+
+use super::error::{GqlError, Result};
+use super::ext::ContextExt;
+use super::models::message::Message;
+use async_graphql::{futures_util::Stream, Context, Object, Subscription};
+use async_stream::stream;
+use chrono::Utc;
+use common::types::chat_message;
+use fred::prelude::PubsubInterface;
+use std::collections::HashMap;
+
+#[derive(Default)]
+pub struct ChatMutation;
+
+#[derive(Default)]
+pub struct ChatSubscription;
+
+#[Object]
+impl ChatMutation {
+    // Send message in chat. You need to be logged in for that.
+    async fn send_message<'ctx>(
+        &self,
+        ctx: &Context<'_>,
+        #[graphql(desc = "ID of chat room where the message will be send.")] chat_id: i64,
+        #[graphql(desc = "Message content that will be published.")] content: String,
+    ) -> Result<Message> {
+        let global = ctx.get_global();
+        let request_context = ctx.get_session();
+        const MAX_MESSAGE_LENGTH: usize = 500;
+
+        if content.len() > MAX_MESSAGE_LENGTH {
+            return Err(GqlError::InvalidInput.with_message("Message too long"));
+        }
+
+        let session = request_context.get_session(global).await?;
+
+        let user_id = session
+            .as_ref()
+            .ok_or_else(|| GqlError::Unauthorized.with_message("You need to be logged in"))?
+            .user_id;
+
+        let user = global
+            .user_by_id_loader
+            .load_one(user_id)
+            .await
+            .map_err_gql("Failed to fetch user")?
+            .ok_or(GqlError::InvalidInput.with_message("User not found"))?;
+
+        let _chat = global
+            .chat_room_by_id_loader
+            .load_one(chat_id)
+            .await
+            .map_err_gql("Failed to fetch chat")?
+            .ok_or(GqlError::InvalidInput.with_message("Chat not found"))?;
+
+        let db_message = sqlx::query_as!(
+            chat_message::Model,
+            "INSERT INTO chat_messages (chat_room_id, author_id, message, created_at) VALUES ($1, $2, $3, $4) RETURNING *",
+            chat_id,
+            user_id,
+            content,
+            Utc::now()
+        )
+        .fetch_one(&*global.db)
+        .await
+        .map_err_gql("Failed to send message")?;
+
+        let message_to_send = Message {
+            message_id: db_message.id,
+            chat_id,
+            username: user.username,
+            content: content.to_string(),
+            message_type: "message".to_string(),
+            metadata: HashMap::from([("color".to_string(), "teal".to_string())]),
+        };
+
+        match global
+            .redis_pool
+            .publish::<String, String, String>(
+                format!("chat:{}", chat_id),
+                serde_json::json!(message_to_send).to_string(),
+            )
+            .await
+        {
+            Ok(_) => {}
+            Err(_) => {
+                return Err(GqlError::InternalServerError.with_message("Failed to publish message"));
+            }
+        };
```

Comment:
Perhaps something like this (untested + unformatted)

```suggestion
        if let Err(err) = global.redis_pool.publish::<String, String, String>(format!("chat:{}", chat_id), serde_json::json!(message_to_send).to_string()).await {
            tracing::error!("failed to publish redis message: {}", err)
            return Err(GqlError::InternalServerError.with_message("Failed to publish message"));
        }
```